src/gtk/toga_gtk/libs/style.py

```diff
@@ -0,0 +1,49 @@
+from . import Gtk
+
+
+def apply(style_context, style):
+    # creating StyleProvider (i.e CssProvider)
+    style_provider = Gtk.CssProvider()
+    style_provider.load_from_data(style.encode())
+
+    # setting the StyleProvider to StyleContext
+    style_context.add_provider(
+        style_provider,
+        Gtk.STYLE_PROVIDER_PRIORITY_USER,
+    )
+    style_context.add_class("custom-style")
```

Comment:
If I'm reading this right, GTK styling is based on the idea that every app has a CSS stylesheet (possibly building on a "system theme" stylesheet); and then individual widgets can tweak that style by attaching an encoded style that defines a class, and attaching that class to the widget.

If that's the case, I have 2 questions/clarifications:

1) Would it make sense to use a name like `toga` as the custom class, rather than the generic class name?I'm thinking mostly for debugging purposes, being able to identify where a style change comes from. `custom-style` is sufficiently generic that it might not be obvious where the style is coming from.

2) Is there any potential for namespace collision here? i.e., if I have 2 labels, both of which have a style set using this technique, but *different* styles (e.g., label 1 is red, label 2 is blue), are we going to see any overlap in the application of styles? What is the scope of a `Gtk.CssProvider`?

Reply:
> If I'm reading this right, GTK styling is based on the idea that every app has a CSS stylesheet (possibly building on a "system theme" stylesheet); and then individual widgets can tweak that style by attaching an encoded style that defines a class, and attaching that class to the widget.

Yes. This is what is  happening.

> Would it make sense to use a name like toga as the custom class, rather than the generic class name?I'm thinking mostly for debugging purposes, being able to identify where a style change comes from. custom-style is sufficiently generic that it might not be obvious where the style is coming from.

I agree with you. This will be done in next commit.

> Is there any potential for namespace collision here? i.e., if I have 2 labels, both of which have a style set using this technique, but different styles (e.g., label 1 is red, label 2 is blue), are we going to see any overlap in the application of styles? What is the scope of a Gtk.CssProvider?

It two labels; label1 and label2 have different style sheets applied to them it will behave as expected. They will have different styles. 
If a css class custom-style is added to a label widget which sets the color to red and size to 20. Later if you add another class with the same name i.e custom-style which sets the color to black. Only the color will change and the size will remain 20.